File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/anim/DefaultNoAnimator.java
package me.yokeyword.fragmentation.anim;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by YoKeyword on 16/2/15.
 */
public class DefaultNoAnimator extends FragmentAnimator implements Parcelable {

    public DefaultNoAnimator() {
        mEnter = 0;
        mExit = 0;
        mPopEnter = 0;
        mPopExit = 0;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
    }

    private DefaultNoAnimator(Parcel in) {
        super(in);
    }

    public static final Creator<DefaultNoAnimator> CREATOR = new Creator<DefaultNoAnimator>() {
        @Override
        public DefaultNoAnimator createFromParcel(Parcel source) {
            return new DefaultNoAnimator(source);
        }

        @Override
        public DefaultNoAnimator[] newArray(int size) {
            return new DefaultNoAnimator[size];
        }
    };
}

File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/anim/DefaultNoAnimator.kt
package me.yokeyword.fragmentation.anim

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by YoKeyword on 16/2/15.
 */
class DefaultNoAnimator : FragmentAnimator, Parcelable {

    constructor() {
        enterAnim = 0
        exitAnim = 0
        popEnterAnim = 0
        popExitAnim = 0
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
    }

    private constructor(`in`: Parcel) : super(`in`)

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<DefaultNoAnimator> = object : Parcelable.Creator<DefaultNoAnimator> {
            override fun createFromParcel(source: Parcel): DefaultNoAnimator {
                return DefaultNoAnimator(source)
            }

            override fun newArray(size: Int): Array<DefaultNoAnimator?> {
                return arrayOfNulls(size)
            }
        }
    }
}
